### PR TITLE
Avoid creating empty templates

### PR DESCRIPTION
**Pull request by hishamco**
Address part of #6684

### CODE REVIEW

**Comment by hishamco** [comment]
@agriffard the issue I'm facing that the page trigger the model error, but when the page loaded the editor displayed without error message

**Comment by agriffard** [comment]
I made a few changes in your PR (to be applied on Edit too).

It is because there was no `<span asp-validation-for="Name"></span>` and because it is a code mirror textarea.

`span` with class `field-validation-error` should be in red in theAdmin css.

**Comment by hishamco** [comment]
>I made a few changes in your PR (to be applied on Edit too).

That's why m next commit refuse ;)

**Comment by hishamco** [comment]
I think this is ready to merge

File: src/OrchardCore.Modules/OrchardCore.Templates/Controllers/TemplateController.cs [hunk 1]
```diff
@@ -128,6 +128,10 @@ public async Task<IActionResult> CreatePost(TemplateViewModel model, string subm
                 {
                     ModelState.AddModelError(nameof(TemplateViewModel.Name), S["The name is mandatory."]);
                 }
+                else if (String.IsNullOrWhiteSpace(model.Content))
+                {
+                    ModelState.AddModelError(nameof(TemplateViewModel.Content), S["The content is mandatory."]);
```
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.Templates/Controllers/TemplateController.cs:133]: you might as well add this validation to the `Edit` action as well

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Templates/Controllers/TemplateController.cs:133]: > I made a few changes in your PR (to be applied on Edit too).

I though Antoine add it ;)

File: src/OrchardCore.Modules/OrchardCore.Templates/Views/Template/Create.cshtml [hunk 2]
```diff
@@ -36,6 +36,7 @@
     <div class="form-group" asp-validation-class-for="Content">
         <label asp-for="Content">@T["Content"]</label>
         <textarea asp-for="Content" rows="10" class="form-control"></textarea>
+        <span asp-validation-for="Content"></span>
```
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.Templates/Views/Template/Create.cshtml:39]: and for `Edit` to?

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Templates/Views/Template/Create.cshtml:39]: > and for Edit to?

> I made a few changes in your PR (to be applied on Edit too).

I will double check this now ...

**Pull request merged**